src/Servers/Kestrel/Core/src/Internal/Infrastructure/TransportManager.cs

```diff
@@ -62,7 +62,9 @@ public async Task<EndPoint> BindAsync(EndPoint endPoint, MultiplexedConnectionDe
                 // TODO Set other relevant values on options
                 var sslServerAuthenticationOptions = new SslServerAuthenticationOptions
                 {
-                    ServerCertificate = listenOptions.HttpsOptions.ServerCertificate
+                    ServerCertificate = listenOptions.HttpsOptions.ServerCertificate,
+                    // TODO: Remove h3-29 after HttpClient updates https://github.com/dotnet/runtime/issues/55894
+                    ApplicationProtocols = new List<SslApplicationProtocol>() { new SslApplicationProtocol("h3"), new SslApplicationProtocol("h3-29") }
```

Comment:
This fix should be available in tomorrow's runtime update.

Reply:
Can this be removed now?

Reply:
Rebased and removed.